src/IIIFPresentation/API/Features/Storage/Requests/GetCollection.cs

```diff
@@ -36,12 +47,21 @@ public async Task<CollectionWithItems> Handle(GetCollection request,
                 cancellationToken);
         }
         
-        IQueryable<Collection>? items = null;
+        List<Collection>? items = null;
+        int total = 0;
 
         if (collection != null)
         {
-            items = dbContext.Collections.Where(s => s.CustomerId == request.CustomerId && s.Parent == collection.Id);
-
+            total = await dbContext.Collections.CountAsync(
```

Comment:
Worth using `Task.WhenAll()` to allow these to run in parallel?

Reply:
I think there are issues with running queries in parallel in EF - https://learn.microsoft.com/en-us/ef/core/dbcontext-configuration/#avoiding-dbcontext-threading-issues